### ISSUE TITLE
Compilation error?

### ISSUE THREAD

**Issue by dirkjanvw** [khodor14/Cdbgtricks#36]
Hi! Curious to try out your tool, but I run into compilation issues.

I get the following compilation issues with tag v0.0.3:
```
$ cmake -S ../ -B . && make
-- The C compiler identification is GNU 13.2.0
-- The CXX compiler identification is GNU 13.2.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /path/to/miniforge3/envs/bio/bin/x86_64-conda-linux-gnu-cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /path/to/miniforge3/envs/bio/bin/x86_64-conda-linux-gnu-c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Configuring done
-- Generating done
-- Build files have been written to: /path/to/bin/Cdbgtricks-v0.0.3/build
[  4%] Building CXX object CMakeFiles/ccdbgupdater.dir/src/main.cpp.o
/path/to/bin/Cdbgtricks-v0.0.3/src/main.cpp:192:6: error: redefinition of 'void show_usage()'
  192 | void show_usage(){
      |      ^~~~~~~~~~
/path/to/bin/Cdbgtricks-v0.0.3/src/main.cpp:17:6: note: 'void show_usage()' previously defined here
   17 | void show_usage(){
      |      ^~~~~~~~~~
/path/to/bin/Cdbgtricks-v0.0.3/src/main.cpp:213:62: error: redefinition of 'std::unordered_map<std::__cxx11::basic_string<char>, std::tuple<bool, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > parseArgs(int, char**)'
  213 | std::unordered_map<std::string,std::tuple<bool,std::string>> parseArgs(int argc,char **argv){
      |                                                              ^~~~~~~~~
/path/to/bin/Cdbgtricks-v0.0.3/src/main.cpp:38:62: note: 'std::unordered_map<std::__cxx11::basic_string<char>, std::tuple<bool, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > parseArgs(int, char**)' previously defined here
   38 | std::unordered_map<std::string,std::tuple<bool,std::string>> parseArgs(int argc,char **argv){
      |                                                              ^~~~~~~~~
make[2]: *** [CMakeFiles/ccdbgupdater.dir/build.make:76: CMakeFiles/ccdbgupdater.dir/src/main.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:134: CMakeFiles/ccdbgupdater.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```

And with the current branch main I get the following:
```
$ cmake -S ../ -B . && make
-- The C compiler identification is GNU 13.2.0
-- The CXX compiler identification is GNU 13.2.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /path/to/miniforge3/envs/bio/bin/x86_64-conda-linux-gnu-cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /path/to/miniforge3/envs/bio/bin/x86_64-conda-linux-gnu-c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Found ZLIB: /path/to/miniforge3/envs/bio/lib/libz.so (found version "1.2.13") 
-- Configuring done
-- Generating done
-- Build files have been written to: /path/to/bin/Cdbgtricks-v0.0.3/build
[  5%] Building CXX object CMakeFiles/cdbgtricks.dir/src/main.cpp.o
tput bel
In file included from /path/to/bin/Cdbgtricks-v0.0.3/src/ParseGFA.h:8,
                 from /path/to/bin/Cdbgtricks-v0.0.3/src/main.cpp:1:
/path/to/bin/Cdbgtricks-v0.0.3/src/unitig.h:19:5: error: 'uint8_t' does not name a type
   19 |     uint8_t unused; //left and write 0000|left|right (left,right={0,1,2,3}=={00,01,10,11})
      |     ^~~~~~~
/path/to/bin/Cdbgtricks-v0.0.3/src/unitig.h:15:1: note: 'uint8_t' is defined in header '<cstdint>'; did you forget to '#include <cstdint>'?
   14 | #include <cmath>
  +++ |+#include <cstdint>
   15 | 
/path/to/bin/Cdbgtricks-v0.0.3/src/unitig.h:23:17: error: 'uint8_t' was not declared in this scope
   23 |   std::vector < uint8_t > unitig_encoding;
      |                 ^~~~~~~
/path/to/bin/Cdbgtricks-v0.0.3/src/unitig.h:23:17: note: 'uint8_t' is defined in header '<cstdint>'; did you forget to '#include <cstdint>'?
/path/to/bin/Cdbgtricks-v0.0.3/src/unitig.h:23:25: error: template argument 1 is invalid
   23 |   std::vector < uint8_t > unitig_encoding;
      |                         ^
/path/to/bin/Cdbgtricks-v0.0.3/src/unitig.h:23:25: error: template argument 2 is invalid
/path/to/bin/Cdbgtricks-v0.0.3/src/unitig.h:33:17: error: expected ')' before 'left_unused'
   33 |   Unitig(uint8_t left_unused, uint8_t right_unused, std::vector < uint8_t > encoding);
      |         ~       ^~~~~~~~~~~~
      |                 )
/path/to/bin/Cdbgtricks-v0.0.3/src/unitig.h:35:3: error: 'uint8_t' does not name a type
   35 |   uint8_t get_left_unused();
      |   ^~~~~~~
/path/to/bin/Cdbgtricks-v0.0.3/src/unitig.h:35:3: note: 'uint8_t' is defined in header '<cstdint>'; did you forget to '#include <cstdint>'?
/path/to/bin/Cdbgtricks-v0.0.3/src/unitig.h:36:3: error: 'uint8_t' does not name a type
   36 |   uint8_t get_right_unused();
      |   ^~~~~~~
/path/to/bin/Cdbgtricks-v0.0.3/src/unitig.h:36:3: note: 'uint8_t' is defined in header '<cstdint>'; did you forget to '#include <cstdint>'?
/path/to/bin/Cdbgtricks-v0.0.3/src/unitig.h:37:3: error: 'uint64_t' does not name a type
   37 |   uint64_t get_ith_mer(const int i,
      |   ^~~~~~~~
/path/to/bin/Cdbgtricks-v0.0.3/src/unitig.h:37:3: note: 'uint64_t' is defined in header '<cstdint>'; did you forget to '#include <cstdint>'?
/path/tobin/Cdbgtricks-v0.0.3/src/unitig.h:39:3: error: 'uint64_t' does not name a type
   39 |   uint64_t get_next_mer(uint64_t mer,
      |   ^~~~~~~~
/path/tobin/Cdbgtricks-v0.0.3/src/unitig.h:39:3: note: 'uint64_t' is defined in header '<cstdint>'; did you forget to '#include <cstdint>'?
/path/to/bin/Cdbgtricks-v0.0.3/src/unitig.h:42:3: error: 'uint64_t' does not name a type
   42 |   uint64_t get_prev_mer(uint64_t mer,
      |   ^~~~~~~~
/path/to/bin/Cdbgtricks-v0.0.3/src/unitig.h:42:3: note: 'uint64_t' is defined in header '<cstdint>'; did you forget to '#include <cstdint>'?
/path/to/bin/Cdbgtricks-v0.0.3/src/unitig.h:49:17: error: 'uint8_t' was not declared in this scope
   49 |   std::vector < uint8_t > get_encoding();
      |                 ^~~~~~~
/path/to/bin/Cdbgtricks-v0.0.3/src/unitig.h:49:17: note: 'uint8_t' is defined in header '<cstdint>'; did you forget to '#include <cstdint>'?
/path/to/bin/Cdbgtricks-v0.0.3/src/unitig.h:49:25: error: template argument 1 is invalid
   49 |   std::vector < uint8_t > get_encoding();
      |                         ^
/path/to/bin/Cdbgtricks-v0.0.3/src/unitig.h:49:25: error: template argument 2 is invalid
make[2]: *** [CMakeFiles/cdbgtricks.dir/build.make:76: CMakeFiles/cdbgtricks.dir/src/main.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:136: CMakeFiles/cdbgtricks.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```

**Comment by khodor14** [comment]
Hi @dirkjanvw
Thanks for your interest on Cdbgtricks. The release that you have used was during production. I just made a new one.

I have just compiled the main branch.
![Screenshot from 2024-12-01 21-01-50](https://github.com/user-attachments/assets/9dcfd203-fbfd-4145-81de-f3c967834fc5)
But I think the compilor on your machine is not recognizing uint8_t which can be fixed by including <cstdint> in the unitig.h file. 
I will create another branch where I do the include and let you know.
